examples/api/java/Combination.java

```diff
@@ -10,90 +10,119 @@
-
-    // Construct the assumptions
-    Expr assumptions =
-      em.mkExpr(Kind.AND,
-                em.mkExpr(Kind.LEQ, zero, f_x), // 0 <= f(x)
-                em.mkExpr(Kind.LEQ, zero, f_y), // 0 <= f(y)
-                em.mkExpr(Kind.LEQ, sum, one),  // f(x) + f(y) <= 1
-                p_0.notExpr(),                  // not p(0)
-                p_f_y);                         // p(f(y))
-    smt.assertFormula(assumptions);
-
-    System.out.println("Given the following assumptions:");
-    System.out.println(assumptions);
-    System.out.println("Prove x /= y is entailed. "
-        + "CVC4 says: " + smt.checkEntailed(em.mkExpr(Kind.DISTINCT, x, y))
-        + ".");
-
-    System.out.println("Now we call checksat on a trivial query to show that");
-    System.out.println("the assumptions are satisfiable: " +
-                       smt.checkSat(em.mkConst(true)) + ".");
-
-    System.out.println("Finally, after a SAT call, we recursively call smt.getValue(...) on " +
-                       "all of the assumptions to see what the satisfying model looks like.");
-    prefixPrintGetValue(smt, assumptions, 0);
+    Term f_x = slv.mkTerm(Kind.APPLY_UF, f, x);
+    Term f_y = slv.mkTerm(Kind.APPLY_UF, f, y);
+    Term sum = slv.mkTerm(Kind.PLUS, f_x, f_y);
+    Term p_0 = slv.mkTerm(Kind.APPLY_UF, p, zero);
+    Term p_f_y = slv.mkTerm(Kind.APPLY_UF, p, f_y);
+
+    // Construct the assertions
+    Term assertions = slv.mkTerm(Kind.AND,
+        new Term[] {
+            slv.mkTerm(Kind.LEQ, zero, f_x), // 0 <= f(x)
+            slv.mkTerm(Kind.LEQ, zero, f_y), // 0 <= f(y)
+            slv.mkTerm(Kind.LEQ, sum, one), // f(x) + f(y) <= 1
+            p_0.notTerm(), // not p(0)
+            p_f_y // p(f(y))
+        });
+    slv.assertFormula(assertions);
+
+    System.out.println("Given the following assertions:\n" + assertions + "\n");
+
+    System.out.println("Prove x /= y is entailed. \n"
+        + "cvc5: " + slv.checkEntailed(slv.mkTerm(Kind.DISTINCT, x, y)) + ".\n");
+
+    System.out.println("Call checkSat to show that the assertions are satisfiable. \n"
+        + "cvc5: " + slv.checkSat() + ".\n");
+
+    System.out.println("Call slv.getValue(...) on terms of interest.");
+    System.out.println("slv.getValue(" + f_x + "): " + slv.getValue(f_x));
+    System.out.println("slv.getValue(" + f_y + "): " + slv.getValue(f_y));
+    System.out.println("slv.getValue(" + sum + "): " + slv.getValue(sum));
+    System.out.println("slv.getValue(" + p_0 + "): " + slv.getValue(p_0));
+    System.out.println("slv.getValue(" + p_f_y + "): " + slv.getValue(p_f_y) + "\n");
+
+    System.out.println("Alternatively, iterate over assertions and call slv.getValue(...) "
+        + "on all terms.");
+    prefixPrintGetValue(slv, assertions);
+
+    System.out.println();
+    System.out.println("You can also use nested loops to iterate over terms.");
+    Iterator<Term> it1 = assertions.iterator();
+    while (it1.hasNext())
```

Comment:
Can we do `for (Term it1: assertions)` here?

Reply:
Yes, which is written few lines later. 
https://github.com/cvc5/cvc5/pull/7225/files#diff-84065c47299f1c133669b2a4c677efbf630b733dc7ec4071368d778256ba517dR119